contracts/kernel/KernelProxy.sol

```diff
@@ -11,14 +11,15 @@ contract KernelProxy is KernelStorage, DelegateProxy {
     * @param _kernelImpl Address of the contract used as implementation for kernel
     */
     function KernelProxy(address _kernelImpl) public {
-        kernelImpl = _kernelImpl;
+        apps[keccak256(CORE_NAMESPACE, KERNEL_APP_ID)] = _kernelImpl;
+        assert(apps[KERNEL_APP] == _kernelImpl);
```

Comment:
Seems like a weird place to check that `keccak256(CORE_NAMESPACE, KERNEL_APP_ID) == KERNEL_APP`. The tests might be better?